### ISSUE TITLE
Publish a new release

### ISSUE THREAD

**Issue by lawnsea** [wycats/handlebars.js#1256]
We would like to use the nested partial blocks fix that landed from #1243. Can we cut a new release, please?

**Comment by nknapp** [comment]
From what I read in the last weeks, there is a urgent shortage of collaborators in this project.
@wycats, @kpdecker, @ErisDS: There is some [documentation about performing the release](https://github.com/wycats/handlebars.js/blob/master/CONTRIBUTING.md#releasing). I have understood that @ErisDS attempted to start maintaining the 3.0-branch, but [took a u-turn](https://github.com/wycats/handlebars.js/commit/83b8e846a3569bd366cf0b6bdc1e4604d1a2077e#commitcomment-17078149) due to the amount of work this means.

My question is: Is the release-documentation up-to-date? Does the generator-release work for the master branch and just not for other branches? Or is there something missing and everything is more complicated? If it is just about performing some releases from time to time, I would offer to help. I probably don't know enough about the code to judge every single PR, but right know a single release would be sufficient. 

This project is heavily used, with 10905 stars and 54,683 downloads a day, three maintainers seem to be a too few, especially if they are all occupied otherwise.

**Comment by ErisDS** [comment]
Doing a 3.0 release was too much work because all of the release tools assume you're releasing master, and I would have needed to release another branch.

As much as I have some familiarity with the code and a great deal of familiarity with the functionality of handlebars, yeoman and the gem-related steps of the release are quite unfamiliar tools to me. I also only have the ability to publish to npm, and not to rubygems.org.

In short, I'm here and could potentially do a release but only half of one 😁

**Comment by dpehrson** [comment]
I am currently blocked hard by #1243, but need the fix published in NPM.

I feel bad pinging this issue since I am not in a position to help solve the problem, but I am looking for guidance on whether I should cut my losses and move on to a different templating solution or if there is any likelihood that a release is pushed before the end of the month.

Thank you all for the hard (and usually thankless) work you do in maintaining a popular open source project, especially dealing with the community-related issues.

**Comment by lawnsea** [comment]
@ErisDS I guess my vote is to just do the npm release and deal with releasing the gem in another issue.

@wycats is there someone who can do the gem release? I would *really* like to get this fix out where we can actually use it.

**Comment by lawnsea** [comment]
To clarify my comment above, the npm release is what we need. We don't use the gem, AFAIK.

**Comment by dpehrson** [comment]
Just to relieve the pressure: This issue is no longer blocking us, we are going to switch to another templating solution.

Thanks to everyone for all their hard work over the years, I apologize that I can't be part of the solution to the problems facing the project at this time.

**Comment by stevenvachon** [comment]
```json
"dependencies": {
  "handlebars": "wycats/handlebars.js"
}
```

**Comment by lawnsea** [comment]
Alright, y'all, @wycats is hooking me up with the permissions to do a release.

Next question: is this a major, minor, or patch release? Based on the [currently unreleased commits on master](https://github.com/wycats/handlebars.js/compare/205c61cfb1acdb599bbdfcf2d356641254e09e5c...master), it seems like this will need to be a major release. Thoughts?

**Comment by ErisDS** [comment]
@lawnsea I didn't do the release because when I pointed out that there were breaking changes and this needs to be a major, I was informed by @wycats that he wasn't comfortable with some of the changes on master. @wycats & @mmun both said they didn't think there should be a major, implying that some changes needed reverting.

**Comment by lawnsea** [comment]
@ErisDS ok, thanks for that detail. As an alternative, I could cut a branch from 4.0.5 and land the non-controversial changes to that. Figuring out which changes are non-controversial will take some time, of course...

**Comment by mmun** [comment]
Would either of you mind making a list of the breaking changes?

**Comment by lawnsea** [comment]
@mmun 
some commits that look breaking to me:
3f77b82
63a0889
8517352
118836f

That list is likely not exhaustive. I doubt that many (any?) users are depending on the stuff removed from the compiler API. I'm more worried about the AMD build artifacts.

Given the number of downstream dependents that will automatically consume a minor or patch release, we should be very careful here.

**Comment by lawnsea** [comment]
Ok, I've confirmed that my patch for #1243 applies cleanly to v4.0.5 and that the tests pass. So, at @wycats suggestion, I propose we cut a release with those patches as v4.0.6 and have a longer discussion about what to do with the rest of master.

If there are bug fix commits on master that y'all think should go in v4.0.6, I'm fine with that as well.

**Comment by lawnsea** [comment]
Unless anyone objects, I'm going to create a 4.x branch from the v4.0.5 tag and open a PR with the bug fixes I propose to ship in v4.0.6.

**Comment by lawnsea** [comment]
Candidate release branch cut. Please take a look at #1270 and let me know if you want changes.

@wycats @ErisDS @mmun

**Comment by lawnsea** [comment]
Just published 4.0.6
